sympy/core/backend.py

```diff
@@ -0,0 +1,27 @@
+import os
+USE_SYMENGINE = os.getenv('USE_SYMENGINE')
+
+if USE_SYMENGINE:
+    import symengine as sm
+    from symengine import (Symbol, Integer, sympify, sympify as S,
+        SympifyError, exp, log, gamma, sqrt, I, E, pi, Matrix,
+        sin, cos, tan, cot, csc, sec, asin, acos, atan, acot, acsc, asec,
+        sinh, cosh, tanh, coth, asinh, acosh, atanh, acoth,
+        Lambdify as lambdify, symarray, diff, zeros, eye, diag, ones, zeros,
+        expand, UndefFunction as Function, symbols, var)
+    from symengine.lib.symengine_wrapper import (Function as FunctionClass,
+        Add as AddClass, Mul as MulClass, add as Add, mul as Mul,
+        FunctionSymbol as AppliedUndef, Derivative as DerivativeClass,
+        Sin as SinClass, Cos as CosClass, Tan as TanClass)
+else:
+    import sympy as sm
```

Comment:
Is this intentional? The one place where you imported `sm` below you imported this module as `sm`, not `sm` from this module. I personally wouldn't have this here. `sm` isn't even the shorthand I'd recommend for SymPy. I always tell people to use `sym`, but it's far from standardized.